### PR TITLE
De-duplicate find user

### PR DESCRIPTION
**Pull request by eikesr**

### CODE REVIEW

**Review by boxdot — APPROVED** [review]

File: gerritbot/src/lib.rs [hunk 1]
```diff
@@ -415,7 +415,7 @@ where
             .chain(std::iter::once(&change.owner))
             .filter(|user| user.is_human())
             .filter_map(|user| user.spark_email())
-            .filter_map(move |email| self.state.find_user_by_email(email))
+            .filter_map(move |email| self.state.find_user(email))
```
Comment (boxdot) [inline gerritbot/src/lib.rs:418]: Apparently now you always return something, so you might want to remove the option here.

**Review by boxdot — APPROVED** [review]

File: gerritbot/src/lib.rs (hunk 1 above)
Comment (eikesr) [reply gerritbot/src/lib.rs:418]: Surprisingly, since both `find_user_by_email` and `find_user` return an option. Will look into the clippy warnings

Comment (boxdot) [reply gerritbot/src/lib.rs:418]: Looks like this warning was already before there. Feel free to ignore.

**Comment by boxdot** [comment]
CI was fixed in #105. Please rebase to get its validation.

**Review by boxdot — APPROVED** [review]

**Pull request merged**